### PR TITLE
Jp/sum spaces beta

### PR DESCRIPTION
**Pull request by ioannisPApapadopoulos**

### CODE REVIEW

File: src/frame.jl [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Custom SVD solver for least squares problems
+function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
+    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
```
Comment (dlfivefifty) [inline src/frame.jl:3]: Why not just do `Matrix(A)`?

File: src/frame.jl [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# Custom SVD solver for least squares problems
+function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
+    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
+    U,S,V = svd(Am)
+    S⁺ = inv.(S)[:]
```
Comment (dlfivefifty) [inline src/frame.jl:5]: Why the `[:]`?

File: src/frame.jl [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# Custom SVD solver for least squares problems
+function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
+    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
+    U,S,V = svd(Am)
```
Comment (dlfivefifty) [inline src/frame.jl:4]: Don't use capital `S` since its a `Vector` (I usually just call it `σ`)

File: src/frame.jl [hunk 4]
```diff
@@ -0,0 +1,111 @@
+# Custom SVD solver for least squares problems
+function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
+    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
+    U,S,V = svd(Am)
+    S⁺ = inv.(S)[:]
+    S⁺[S⁺ .> 1/tol] .= 0
```
Comment (dlfivefifty) [inline src/frame.jl:6]: I probably would have done something like the following:
```julia
U,σ,V = svd(Am)
filter!(>(tol), σ)
r = length(σ)
V[:,1:r] * (inv.(σ) .* (U[:,1:r]' * b))
```
But it doesn't matter too much.

**Comment by ioannisPApapadopoulos** [comment]
Ok I have made the changes, let me know if you're happy for me to merge to main.

**Pull request merged**